core/wasm/runtest/src/lib.rs

```diff
@@ -43,27 +48,30 @@ impl External for MyExt {
         _mana: u32,
         _amount: u64,
     ) -> ExtResult<PromiseId> {
-        Err(ExtError::NotImplemented)
+        self.num_receipts += 1;
+        Ok(PromiseId::Receipt(generate_promise_id(self.num_receipts - 1)))
     }
 
     fn promise_then(
         &mut self,
-        _promise_id: PromiseId,
+        promise_id: PromiseId,
         _method_name: Vec<u8>,
         _arguments: Vec<u8>,
         _mana: u32,
     ) -> ExtResult<PromiseId> {
-        Err(ExtError::NotImplemented)
-    }
-
-    fn promise_and(
-        &mut self,
-        _promise_id1: PromiseId,
-        _promise_id2: PromiseId,
-    ) -> ExtResult<PromiseId> {
-        Err(ExtError::NotImplemented)
+        match promise_id {
+            PromiseId::Receipt(_) => {
+                assert!(false);
```

Comment:
Why do we need this assert? It already returns an error.

Reply:
Easier to debug.